packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -150,12 +143,30 @@ async function prototypeChanges(
     outdir: path.join(process.cwd(), 'original'),
   });
 
-  const cx = await toolkit.synth(source);
-  const assembly = cx.cloudAssembly;
-
   const updateObj: Record<string, boolean> = {};
-  updateObj[flagName] = boolValue;
+  if (typeof (flagNames) == 'string') {
+    if (baseContextValues[flagNames] == value) {
+      await ioHelper.defaults.error('Flag is already set to the specified value. No changes needed.');
+      return false;
+    }
+    updateObj[flagNames] = value!;
+  } else {
+    if (recommended) {
+      for (const flagName of flagNames) {
+        const flag = flagData.find(f => f.name === flagName);
+        const boolValue = toBooleanValue(flag!.recommendedValue);
```

Comment:
Are you sure `flag` is defined here? The flag name is free form set by the user, right?